contracts/pallet-ibc/src/impls.rs

```diff
@@ -66,30 +66,27 @@ where
 		ctx: &mut Context<T>,
 		messages: Vec<ibc_proto::google::protobuf::Any>,
 	) {
-		let (events, logs, errors) = messages.into_iter().fold(
-			(vec![], vec![], vec![]),
-			|(mut events, mut logs, mut errors), msg| {
+		let (events, logs) =
+			messages.into_iter().fold((vec![], vec![]), |(mut events, mut logs), msg| {
 				match ibc::core::ics26_routing::handler::deliver(ctx, msg) {
 					Ok(MsgReceipt { events: temp_events, log: temp_logs }) => {
-						events.extend(temp_events);
+						events.extend(temp_events.into_iter().map(Ok));
 						logs.extend(temp_logs);
 					},
-					Err(e) => errors.push(e),
+					Err(e) => {
+						log::trace!(target: "pallet_ibc", "execution error: {}", e);
+						events.push(Err(e));
+					},
 				}
-				(events, logs, errors)
-			},
-		);
+				(events, logs)
+			});
 
 		log::trace!(target: "pallet_ibc", "logs: {:#?}", logs);
-		log::trace!(target: "pallet_ibc", "errors: {:#?}", errors);
```

Comment:
Still log errors

Reply:
What do you mean?

Reply:
Don't remove the error logging, it's still useful